relay-general/src/protocol/contexts/device.rs

```diff
@@ -160,6 +160,10 @@ pub struct DeviceContext {
     /// Whether location support is available on the device.
     pub supports_location_service: Annotated<bool>,
 
+    // The performance class of the device, stored as a number.
+    // This value is synthesized from the device's specs in normalize_device_context.
+    pub class: Annotated<u64>,
```

Comment:
Since the device class is an enumeration with semantic values, can we make it an `enum` instead? For instance, what should happen when an SDK sends the value 42? 

If you want to allow any number, you can introduce a so-called "newtype":

```rust
#[derive(Clone, Copy, Debug, Deserialize, Eq, PartialEq Serialize)]
struct DeviceClass(pub u64);

impl DeviceClass {
    const Low: Self = Self(1);
    const Medium: Self = Self(1);
    const High: Self = Self(1);
}

// use as `DeviceClass::Low`
```

Reply:
Hey Jan, I went with the newtype that you suggested!
We have plans to support more device class values in the future and theres use in being able to compare the number values so I think this works best for us.